zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/JobState.java

```diff
@@ -37,6 +51,9 @@ void forEachActivatableJobs(
 
   long findBackedOffJobs(final long timestamp, final BiPredicate<Long, JobRecord> callback);
 
+  /** Index to point to a specific position in the jobs with deadline column family. */
+  record Index(long key, long deadline) {}
```

Comment:
:+1: Nice that you added a type for that!

:wrench: Since `JobState` tracks more than just deadlines, I'd suggest to rename it to `DeadlineIndex` or something similar.